### PR TITLE
Add viem

### LINKED ISSUE #33: Remove ethers.js for viem
- Blocked by wagmi@1.0.0 release

### PR DESCRIPTION
**Pull request by janndriessen**
**Description**

* Adds [viem](https://viem.sh) 
* Updates wagmi and rainbowkit to latest versions
* Removes ethers

**Tests**

Changes didn't affect any tests.

starter

**Metadata**

- Fixes #33

### CODE REVIEW

**Comment by janndriessen** [comment]
Created in anticipation of the ETHGlobal Superhack next week where I'd love to use this template but with `viem`. If it's not merged maybe it could be an extra branch.

File: .env.example [hunk 1]
```diff
@@ -20,3 +20,7 @@ ANVIL_CHAIN_ID=31337
 
 # Used to fork mainnet for tests
 ANVIL_FORK_URL=https://mainnet.optimism.io
+
+# WalletConnect v2 requires a project ID. You can obtain it from your WC dashboard:
+# https://docs.walletconnect.com/2.0/web/web3wallet/installation#obtain-project-id
```
Comment (roninjin10) [inline .env.example:25]: Let's add this to the Readme around the spot where we mention the etherscan key

**Review by roninjin10 — CHANGES_REQUESTED** [review]
Amazing thank you so much for this pr and I'm excited to see you are planning to use it to hack with in an upcoming hackathon. Just 2 minor requests:
1. Screenshot the app running correctly
2. Let's update the README.md about how to get the wallet connect key

**Comment by janndriessen** [comment]
> Amazing thank you so much for this pr and I'm excited to see you are planning to use it to hack with in an upcoming hackathon. Just 2 minor requests:
> 
> 1. Screenshot the app running correctly
> 2. Let's update the README.md about how to get the wallet connect key

 🤝  Updated as requested.

**Review by roninjin10 — APPROVED** [review]

**Pull request merged**